### ISSUE TITLE
XGBRFRegressor - support or deprecate early stopping please

### ISSUE THREAD

**Issue by rwjmiller** [dmlc/xgboost#7144]
In the python scikit-learn interface to XGBoost, we have XGBRFRegressor. When fitting one of these, it is possible to specify all the parameters relating to early stopping:

- eval_set
- eval_metric
- early_stopping_rounds
- sample_weight_eval_set
- base_margin_eval_set

However, because `num_boost_round` is overridden to 1 for XGBRFRegressor, none of these actually do anything very useful, and the lack of any warning message when attempting to use them can be a bit confusing. It would be nice if we could either:

- Grow trees one at a time (without gradient boosting) and apply early stopping on that basis, or
- Deprecate these parameters for XGBRFRegressor.fit()  to make people aware that this functionality is not supported, and maybe also make this point a bit clearer in the docs.

**Comment by trivialfis** [comment]
Thanks for raising the issue.  I think the best path for us to proceed is to throw an error for now and then add proper random forest training to our roadmap for prioritization.
